ringo_printtemplate/odfconv.py

```diff
@@ -67,6 +68,10 @@ def convert(self, data, format="ods", update=True):
         :returns: Converted data.
 
         """
+        # check document type before writing or opening
+        mimetype = check_mime_type_from_buffer(data)
+        if mimetype != "application/vnd.oasis.opendocument.text":
+            return None
```

Comment:
I mean_render_template. (Sorry for misdirecting you) This method gets called from print and print_template and already has a check on the mimetype. If the check fails, the render_template methods returns None which is handled by the render_final_response method by raising an exception. I do not see the need for further checks.